src/prover/mod.rs

```diff
@@ -7,9 +7,631 @@
 //! APIs are designed to be imported for use qualified (e.g.
 //! `batch_insertion::Prover`, `batch_insertion::Identity` and so on).
 
-pub mod batch_insertion;
+pub mod identity;
 pub mod map;
 pub mod proof;
 
+use std::collections::HashSet;
```

Comment:
I'm assuming not much changed in this file from it's previous version under `batch_insertion/mod.rs`, right?

Reply:
Yes, the functionality did not change. I moved the logic within `batch_insertion/mod.rs` into `prover/mod.rs`.